js/src/components/tree-select-control/options.test.js

```diff
@@ -18,37 +18,46 @@ const options = [
 			{ value: 'IT', label: 'Italy' },
 		],
 	},
-	{
-		value: 'NA',
-		label: 'North America',
-		children: [
-			{ value: 'US', label: 'United States' },
-			{ value: 'CA', label: 'Canada' },
-		],
-	},
-	{
-		value: 'AS',
-		label: 'Asia',
-	},
 ];
 
 describe( 'TreeSelectControl - Options Component', () => {
 	it( 'Expands and collapses groups', () => {
-		const { queryByText } = render( <Options options={ options } /> );
-
-		options.forEach( ( option ) => {
-			const optionItem = queryByText( option.label );
-			expect( optionItem ).toBeTruthy();
-			options.children?.forEach( ( child ) => {
-				const childItem = queryByText( child.label );
-				expect( childItem ).toBeFalsy();
-			} );
-
-			fireEvent.click( optionItem );
-			options.children?.forEach( ( child ) => {
-				const childItem = queryByText( child.label );
-				expect( childItem ).toBeTruthy();
-			} );
+		const { queryAllByRole, queryByText, queryByRole } = render(
+			<TreeSelectControl options={ options } />
+		);
+
+		const control = queryByRole( 'combobox' );
+		fireEvent.click( control );
+
+		const optionItem = queryByText( 'Europe' );
+		const option = options[ 0 ];
+		expect( optionItem ).toBeTruthy();
+
+		option.children.forEach( ( child ) => {
+			const childItem = queryByText( child.label );
+			expect( childItem ).toBeFalsy();
+		} );
+
+		const button = queryAllByRole( 'button' );
+		fireEvent.click( button[ 0 ] );
+
+		option.children.forEach( ( child ) => {
+			const childItem = queryByText( child.label );
+			expect( childItem ).toBeTruthy();
+		} );
+
+		fireEvent.click( button[ 0 ] );
+
+		option.children.forEach( ( child ) => {
+			const childItem = queryByText( child.label );
+			expect( childItem ).toBeFalsy();
+		} );
+
+		fireEvent.click( optionItem );
+
+		option.children.forEach( ( child ) => {
+			const childItem = queryByText( child.label );
+			expect( childItem ).toBeTruthy();
```

Comment:
💅  
Could be separated into another test case to make it more atomic, or maybe add a code comment for indicating this part is testing when the parent is checked, the children should be expanded.